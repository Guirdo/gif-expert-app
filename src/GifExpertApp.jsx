import React, { useState } from 'react'
import AddCategory from './Components/AddCategory';
import { GifGrid } from './Components/GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories);

    return (
        <div className="m-5">
            <div className="head-app">
                <h2>GifExpertApp</h2>
                <AddCategory setCategories={setCategories} />
                <hr />
            </div>

            <ol>
                {
                    categories.map(c =>
                        <GifGrid
                            category={c}
                            key={c}
                        />
                    )
                }
            </ol>

        </div>
    )
}

export default GifExpertApp
