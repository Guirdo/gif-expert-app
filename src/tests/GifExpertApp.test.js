import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Testing GifExpertApp',()=>{

    test('should render the component correctly',()=>{
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    });

    test('should show a category list',()=>{
        const categories = ['Simpson','Goku'];

        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });

});