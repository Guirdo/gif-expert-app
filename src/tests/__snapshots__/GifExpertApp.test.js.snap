// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifExpertApp should render the component correctly 1`] = `
<div
  className="m-5"
>
  <div
    className="head-app"
  >
    <h2>
      GifExpertApp
    </h2>
    <AddCategory
      setCategories={[Function]}
    />
    <hr />
  </div>
  <ol />
</div>
`;

exports[`Testing GifExpertApp should show a category list 1`] = `
<div
  className="m-5"
>
  <div
    className="head-app"
  >
    <h2>
      GifExpertApp
    </h2>
    <AddCategory
      setCategories={[Function]}
    />
    <hr />
  </div>
  <ol>
    <GifGrid
      category="Simpson"
      key="Simpson"
    />
    <GifGrid
      category="Goku"
      key="Goku"
    />
  </ol>
</div>
`;
